### PR TITLE
added new object property has roost for issue #453

### PR DESCRIPTION
**Pull request by diatomsRcool**
Added new object property "has roost" as child of "has habitat" to address issue #453

### CODE REVIEW

File: src/ontology/ro-edit.owl [hunk 1]
```diff
@@ -5828,6 +5829,16 @@ AnnotationAssertion(obo:IAO_0000115 obo:RO_0008508 "inverse of lays eggs on")
 AnnotationAssertion(rdfs:label obo:RO_0008508 "has eggs laid on by")
 SubObjectPropertyOf(obo:RO_0008508 obo:RO_0002619)
 
+# Object Property: obo:RO_0008509 (has roost)
+
+AnnotationAssertion(obo:IAO_0000115 obo:RO_0008509 "x 'has roost' y if and only if: x is an organism, y is a habitat, and y can support rest behaviors x.")
+AnnotationAssertion(dce:creator obo:RO_0008509 <http://orcid.org/0000-0002-2908-3327>)
+AnnotationAssertion(dce:date obo:RO_0008509 "2023-01-18T14:28:21Z"^^xsd:dateTime)
+AnnotationAssertion(oboInOwl:inSubset obo:RO_0008509 subsets:ro-eco)
+AnnotationAssertion(rdfs:comment obo:RO_0008509 "A population of xs will possess adaptations (either evolved naturally or via artifical selection) which permit it to rest in y.")
```
Comment (cthoyt) [inline src/ontology/ro-edit.owl:5838]: can you please give a concrete example with terms in existing OBO ontologies as subject/predicate annotated with IAO:0000112 (example of usage)

Comment (wdduncan) [reply src/ontology/ro-edit.owl:5838]: @cthoyt Do you mean just an example of how the term is used? E.g., "bird has roost some nest"

Comment (cthoyt) [reply src/ontology/ro-edit.owl:5838]: yeah, that would be great, as long as it gives explicit references to Bird and Nest using OBO PURLs

**Comment by wdduncan** [comment]
@diatomsRcool I added `habitat` as the range of `has habitat`. This change should propagate down to `has roost`. Can you confirm the range should be `habitat`?

I also change the `dc:creator` annotation to `dcterms:contributor`. @cthoyt Is this the current protocol for term attribution.

**Comment by cthoyt** [comment]
I'm not sure if we have a concrete choice on which predicate should be used, other than it should be DCTERMS and not DCE. Contributor is probably fine!

File: src/ontology/ro-edit.owl (hunk 1 above)
Comment (wdduncan) [reply src/ontology/ro-edit.owl:5838]: I don't think examples of usage have to use obo purls. Free text is fine.  The purpose of "example of usage" is to provide extra context to disambiguate the term.

**Comment by wdduncan** [comment]
Thanks @cthoyt  
@diatomsRcool you can change to `dcterms:creator` if you wish. I was going off my experience with uberon.

File: src/ontology/ro-edit.owl (hunk 1 above)
Comment (wdduncan) [reply src/ontology/ro-edit.owl:5838]: > This is for the benefit of future users, who without a concrete example might not be able to figure out how to use the relationship

I agree. That is the purpose of the annotation. We haven't required that OBO PURLs be used. See examples for `example of usage` in [adjacent to](http://purl.obolibrary.org/obo/RO_0002220), [capable of](http://purl.obolibrary.org/obo/RO_0002215), [characteristic of](http://purl.obolibrary.org/obo/RO_0000052), and many more ...

But @diatomsRcool if you can provide an example OBO PURLs, that would be nice.

**Review by wdduncan — APPROVED** [review]

**Comment by wdduncan** [comment]
@cthoyt @diatomsRcool I think this is ready to merge. Please let me know if you disagree.

**Comment by cthoyt** [comment]
@diatomsRcool yes, that's perfect! You can annotate that on to the property with http://purl.obolibrary.org/obo/IAO_0000112

**Comment by wdduncan** [comment]
@diatomsRcool I will merge after you provide the [example of usage](http://purl.obolibrary.org/obo/IAO_0000112).

File: src/ontology/ro-edit.owl [hunk 2]
```diff
@@ -17,38 +18,35 @@ Prefix(oboInOwl:=<http://www.geneontology.org/formats/oboInOwl#>)
...
-Import(<http://purl.obolibrary.org/obo/ro/imports/envo_import.owl>)
-Import(<http://purl.obolibrary.org/obo/ro/imports/go_import.owl>)
-Import(<http://purl.obolibrary.org/obo/ro/imports/omo_import.owl>)
-Import(<http://purl.obolibrary.org/obo/ro/imports/orcidio_import.owl>)
-Import(<http://purl.obolibrary.org/obo/ro/imports/other_import.owl>)
-Import(<http://purl.obolibrary.org/obo/ro/imports/pato_import.owl>)
-Import(<http://purl.obolibrary.org/obo/ro/imports/uberon_import.owl>)
-Import(<http://purl.obolibrary.org/obo/ro/imports/obi_import.owl>)
-Annotation(dc:description "The OBO Relations Ontology (RO) is a collection of OWL relations (ObjectProperties) intended for use across a wide variety of biological ontologies."@en)
+Import(<http://purl.obolibrary.org/obo/ro/annotations.owl>)
```
Comment (cthoyt) [inline src/ontology/ro-edit.owl:21]: It would be important to resolve all of these spurious changes before finishing the PR

Comment (anitacaron) [reply src/ontology/ro-edit.owl:21]: I'll update the branch from master.

Comment (diatomsRcool) [reply src/ontology/ro-edit.owl:21]: I have no idea why that happened

Comment (anitacaron) [reply src/ontology/ro-edit.owl:21]: This branch started before we changed the repository to use the ODK structure.

**Comment by wdduncan** [comment]
@cthoyt Yes ... I did :( 

I think the the violation concerns the use of dc elements.
image

Here is the sparql.
image

**Pull request merged**